### DISCUSSION TITLE
Auto Scroll / Middle Click Scroll

### DISCUSSION THREAD

**Discussion by 4evy** [Eloston/ungoogled-chromium#1733] [Ideas]
Hey, I think would be awesome if Ungoogled Chromium adds support for **Auto scroll/Middle Click Scroll**. At the moment this feature only exists on Windows, and its hard coded, so it can’t be disabled even if you wanted to. If a user on Linux or macOS wants to use **Auto scroll/Middle Click Scroll** they have to resort to using an extension which only solves the issue to an extend and it’s not a proper solution, it also poses privacy and security concerns. 

This feature **could’ve been added 12 years ago** by the Chromium developers but they never gave a good reason why they couldn’t or didn’t want to do it. Here is the bug report about this issue on chromium site: https://bugs.chromium.org/p/chromium/issues/detail?id=17689

This feature has existed on Firefox for many years and I hope it can be added to Ungoogled Chromium too since it **makes the browsing experience on macOS and Linux much better**.

**Comment by Ahrotahn** [comment]
Autoscroll can be enabled with:
```diff
--- a/third_party/blink/renderer/core/exported/web_view_impl.cc
+++ b/third_party/blink/renderer/core/exported/web_view_impl.cc
@@ -1793,9 +1793,7 @@
       prefs.default_maximum_page_scale_factor);
 #endif
 
-#if defined(OS_WIN)
   RuntimeEnabledFeatures::SetMiddleClickAutoscrollEnabled(true);
-#endif
 
   RuntimeEnabledFeatures::SetTranslateServiceEnabled(
       prefs.translate_service_available);
```

But doesn't look as nice and it would interfere with middle-click paste as mentioned in that bug report.  

I've found it easier to use this AutoScroll extension [Store](https://chrome.google.com/webstore/detail/autoscroll/occjjkgifpmdgodlplnacmkejpdionan) : [Source](https://github.com/Pauan/AutoScroll), since that fixes the issues with the built-in version as well as allows for setting different speeds.

**Reply by 4evy** [reply]
Couldn't there just be a toggle in the settings menu or in flags page? It can be disabled by default and people who want it can go and enable it, just like Firefox.

**Reply by networkException** [reply]
As the extension is an option and such a flag would interfere with other browser functionality I would rather not have it added

**Reply by 4evy** [reply]
> As the extension is an option and such a flag would interfere with other browser functionality I would rather not have it added

Couldn't it be simply a toggle in the settings page then? It could be disabled by default plus this would remove the need for an extension.

**Reply by networkException** [reply]
If it were to work perfectly out of the box thats fine but in the current state (with interfering middle click paste and not looking great) I wouldn't want to ship it. Also toggles rarely (perhaps never?) get added to the settings page, if at all then to the flags.

**Reply by 4evy** [reply]
> with interfering middle click paste and not looking great

I mean again this could be a toggle in the settings ( just like "Show bookmarks bar" ). Which would be disabled by default, thus a user would have to go out of their way to enable it. Even if it's not working that great under Linux, it wouldn't matter that much since it's turn off by default and only people who go out of their way to enable it would notice anything.

As far as I know Middle Click does nothing on macOS, so there shouldn't be a problem there.

> perhaps never?

I mean the "Show bookmarks bar" is a toggle in the settings page.
